src/plugins/chapters/index.js

```diff
@@ -0,0 +1,195 @@
+        this.setupControlBarChapter();
+      });
+    } else {
+      const textTrack = this.player.addTextTrack('chapters');
+      const end = this.player.duration();
+      const cues = Object.entries(this.options).map((entry, index, obj) => {
+        const cue = new VTTCue(
+          parseFloat(entry[0]),
+          parseFloat(obj[index + 1] ? obj[index + 1][0] : end),
+          entry[1]
+        );
+        textTrack.addCue(cue);
+      });
+      this.chaptersTrack = textTrack;
+      this.setupProgressBarMarkers();
+      this.setupProgressBarChapter();
+      this.setupControlBarChapter();
+    }
+  };
+
+  /**
+   * Setup the controlbar chapter display.
+   */
+  ChaptersPlugin.prototype.setupControlBarChapter = function setupControlBarChapter() {
+    const controlBarChapterHolder =
+      this.player.$('.vjs-control-bar-chapter-display') || document.createElement('div');
+    controlBarChapterHolder.setAttribute('class', 'vjs-control-bar-chapter-display');
+
+    const wrapper = this.player.$('.vjs-control-bar .vjs-spacer');
+    wrapper.innerHTML = '';
+    wrapper.classList.add('vjs-control-bar-chapter-wrapper');
+    wrapper.appendChild(controlBarChapterHolder);
+
+    this.chaptersTrack.addEventListener('cuechange', () => {
+      controlBarChapterHolder.innerHTML =
+        this.chaptersTrack.activeCues_.length > 0 ? this.chaptersTrack.activeCues_[0].text : '';
+    });
+  };
+
+  /**
+   * Setup the progress bar markers.
+   */
+  ChaptersPlugin.prototype.setupProgressBarMarkers = function setupProgressBarMarkers() {
+    const total = this.player.duration();
+    const { seekBar } = this.player.controlBar.progressControl;
+    this.chaptersTrack.cues_.forEach(marker => {
+      if (marker.startTime !== 0) {
+        const markerTime = marker.startTime;
+        const left = (markerTime / total) * 100 + '%';
+
+        const markerEl = videojs.dom.createEl('div', undefined, {
+          class: 'vjs-chapter-marker',
+          style: `left: ${left}`
+        });
+
+        seekBar.el().append(markerEl);
+      }
+    });
+  };
+
+  /**
+   * Setup the progrees bar on-hover chapter display.
+   */
+  ChaptersPlugin.prototype.setupProgressBarChapter = function setupProgressBarChapter() {
+    const chapterEl = videojs.dom.createEl('div', undefined, {
+      class: 'vjs-chapter-display',
+      style: `width: ${this.player.$('.vjs-vtt-thumbnail-display').style.width || '160px'}`
+    });
+
+    const timeTooltip = this.player.getDescendant([
+      'controlBar',
+      'progressControl',
+      'seekBar',
+      'mouseTimeDisplay',
+      'timeTooltip'
+    ]);
+
+    timeTooltip.el().parentElement.prepend(chapterEl);
+
+    const getChapterFromPoint = point => {
+      const total = this.player.duration();
+      const seekBarTime = point * total;
+      const chapter = this.chaptersTrack.cues_.find(marker => {
+        return marker.startTime <= seekBarTime && marker.endTime >= seekBarTime;
```

Comment:
Very subjective, but if you ask me this becomes much easier to read, it becomes clear that you are making sure that seekBarTime is between startTime and endTime 
```suggestion
        return seekBarTime >= marker.startTime  &&  seekBarTime  <= marker.endTime;
```